src/Chunk.cpp

```diff
@@ -2076,36 +2074,73 @@ bool cChunk::DoWithEntityByID(UInt32 a_EntityID, cLambdaEntityCallback a_Callbac
 
 
 
-bool cChunk::ForEachBlockEntity(cBlockEntityCallback & a_Callback)
+// Helper function - there's probably a better place for this
+
+template <BLOCKTYPE, BLOCKTYPE...> bool IsOneOfImpl(BLOCKTYPE, std::false_type);
+template <class = void> bool IsOneOfImpl(BLOCKTYPE, std::true_type);
```

Comment:
It needs to be callable with empty template arguments like `IsOneOfImpl<>` so it has to be a template with defaulted arguments.  The choice of `class = void` over an empty `BLOCKTYPE ...` just ensures that if `IsOneOf` dispatches to the wrong overload it is always a compile error.

It's definitely ugly but if SFINAE doesn't work properly in msvc then I don't see a better way.